### ISSUE TITLE
@Lenses macro doesn't support companions that extend traits

### ISSUE THREAD

**Issue by refried** [julien-truffaut/Monocle#205]
```scala
import monocle.macros.Lenses
@Lenses case class Position(x: Int, y: Int)

trait Boom
object Position extends Boom 
```
`[error] /private/tmp/monocle/Foo.scala:2: Invalid annotation target: must be a case class`

There's just no matching case for a companion that extends something, in the expression for `result` in `LensesImpl`.

**Comment by refried** [comment]
I'm guessing that something like this will work as a replacement for the second case, but I can't try it right this second:
```scala
case (classDef @ q"$mods class $tpname[..$tparams] $ctorMods(...$paramss) extends { ..$earlydefns } with ..$parents { $self => ..$stats }")
  :: q"object $objName extends { ..$objEarlyDefs } with ..$objParents { $objSelf => ..$objDefs }"
  :: Nil if mods.hasFlag(Flag.CASE) =>
  q"""
   $classDef
   object $objName extends { ..$objEarlyDefs} with ..$objParents { $objSelf =>
     ..${lenses(tpname, tparams, paramss)}
     ..$objDefs
   }
   """
```

**Comment by julien-truffaut** [comment]
Good catch! Tell me if you have time to give it a go otherwise I will try your fix

**Comment by refried** [comment]
I'm gonna try it now.

**Comment by refried** [comment]
Hoping you won't mind putting out a bugfix release with this :-)

**Comment by julien-truffaut** [comment]
Thanks, I just released 1.1.1

**Comment by refried** [comment]
Great, thanks.
